contributor_docs/fes_reference_dev_notes.md

```diff
@@ -0,0 +1,456 @@
+```javascript
+p5.newObject = function(parameter) {
+   this.parameter = 'some parameter';
+   this.name = 'p5.newObject';
+};
+````
+* Inline documentation: allowed parameter types are `Boolean`, `Number`, `String`, and name of the object (see the above bullet point). Use `Array` for any types of Array parameters. If needed, explain what kind of the specific types of array parameter are allowed (e.g. `Number[]`, `String[]`) in the description section.
+* Currently supported class types (have their `name` parameter): `p5.Color`, `p5.Element`, `p5.Graphics`, `p5.Renderer`, `p5.Renderer2D`, `p5.Image`, `p5.Table`, `p5.TableRow`, `p5.XML`, `p5.Vector`, `p5.Font`, `p5.Geometry`, `p5.Matrix`, `p5.RendererGL`.
+
+#### Performance Issue with the FES
+
+By default, FES is enabled for p5.js, and disabled in `p5.min.js` to prevent FES functions slowing down the process. The error checking system can significantly slow down your code (up to ~10x in some cases). See the [friendly error performance test](https://github.com/processing/p5.js-website/tree/main/src/assets/learn/performance/code/friendly-error-system/).
+
+You can disable this with one line of code at the top of your sketch:
+
+```JavaScript
+p5.disableFriendlyErrors = true; // disables FES
+
+function setup() {
+  // Do setup stuff
+}
+
+function draw() {
+  // Do drawing stuff
+}
+```
+
+Note that this will disable the parts of the FES that cause performance slowdown (like argument checking). Friendly errors that have no performance cost (like giving an descriptive error if a file load fails, or warning you if you try to override p5.js functions in the global space), will remain in place.
+
+
+## Known Limitations
+
+* FES may still result in false negative cases. These are usually caused by the mismatch between designs of the  functions (e.g. drawing functions those are originally designed to be used interchangeably in both 2D and 3D settings) with actual usage cases. For example, drawing a 3D line with
+```JavaScript
+const x3; // undefined
+line(0, 0, 100, 100, x3, Math.PI);
+```
+will escape FES, because there is an acceptable parameter pattern (`Number`, `Number`, `Number`, `Number`) in `line()`'s inline documentation for drawing in 2D setting. This also means the current version of FES doesn't check for the environmental variables such as `_renderer.isP3D`.
+
+* FES is only able to detect global variables overwritten when declared using `const` or `var`. If `let` is used, they go undetected. This is not currently solvable due to the way `let` instantiates variables.
+
+* The functionality described under **`fesErrorMonitor()`** currently only works on the web editor or if running on a local server. For more details see [this](https://github.com/processing/p5.js/pull/4730).
+
+* The extracting variable/function names feature of FES's `sketch_reader` is not perfect and some cases might go undetected (for eg: when all the code is written in a single line).
+
+## Thoughts for Future Work
```

Comment:
great to have all of this documented!

Reply:
I understood this line as "extracting variable/function names from the user code" the feature may miss some cases due to coding style. I will re-write this so it will be clearer.

Reply:
I will add an item to the Thoughts for Future Work section about identifying these cases and update the reference.